docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,757 @@
+- Add to gateway.go the code that connects GR to ovn_cluster_router using the peer ports
+- Change egressip.go so routes are configured at ovn_cluster_router instead of gateway router.
+- Change the ZoneInterConnect handler to be able to add the transit router remote ports to ovn_cluster_router
+- In general re-use as much Layer3 code as possible since these makes Layer2 topology similar to it.
+
+#### At ovnkube-control-plane
+
+At cluster manager at the zuneCluster controller calculate the peers node subnet
+and annotate the node.
+
+To calculate the peer node subnet we have two options:
+- Derive it from the node-id (same we do for transit switch addrs):
+  - good:
+    - No need to maintain the allocator lifecycle
+    - It's possible to entertain the idea of not annotating the node since the source of truth is the node-id
+  - bad:
+    - Add complexity in the form of node-id to subnet derivation code
+    - Make the transit peers node subnet dependant on node-id
+- Create a new subnet allocator that will be re-sync on restarts and allocate 
+  the subnet with it:
+  - good:
+    - Is simpler since it's matter of a new subnet allocator
+    - The transit peers node subnet is no longer dependant on node-id
+  - bad:
+    - Need to maintain the allocator lifecycle
+    - Use more memory
+
+#### Gateway reconciliation
+
+After this topology is configured ovn-kubernetes need to make sure that the new node agnostic gateway are
+configured and that the old gateway is gone.
+
+For IPv4 a garp need to be generated with the gateway from ovn_cluster_router this will overwrite 
+the mac address from the .1 address at the neighbors cache.
+
+For IPv6 two maint things need to be done
+- Send a unsolicited RA to configure the ovn_cluster_router gateway
+- Send one unsolicited RA per node with lifetime 0 to remove the gateways from GRs.
+
+#### Upgrade cleanups
+
+The implementation should handle the ports, routes, policies and NAT that is no longer
+at the gateway router or layer2 switch and remove it so there is nothing dangling from previous
+layer2 topology.
+
+### Testing Details
+
+* Unit test checking topology will need to be adapted.
+* E2e tests should be the same and pass since this is a refactoring not adding or removing features.
+* Scale tests to check if adding the transit router affect it
+* Upgrade tests is the most important in this case, it should do an upgrade
+  while using all the ovn-kuberntes layer2 topology featues and check that they continue working.
+
+### Documentation Details
+
+N/A
+
+## Risks, Known Limitations and Mitigations
+
+This topology move the NAT that masquerades the management port traffic from
+gateway router to ovn_cluster_router so it may disrupt the readiness probes (api server access)
+or name resolution (dns service access), also local gateway traffic may affected.
+
+This topology is going to be use 99% by virtual machines and kubevirt 
```

Comment:
s/use/used

99%?  Do you mean to say something like "However, this topology is going to be used mainly by virtual machines and kubevirt..." ?

Reply:
make sense